### PR TITLE
feat: support to read tinyint

### LINKED ISSUE #17: Tinyint support
Read tinyint to Arrow Integer8 datatype

### PR DESCRIPTION
**Pull request by WenyXu**
Support to read tinyint data

close #17

### CODE REVIEW

**Review by Jefffrey — COMMENTED** [review]
I think TinyInt should map to i8 and not u8

Can see Arrow cpp reference here maps it to Int8 datatype:

https://arrow.apache.org/docs/cpp/orc.html

File: src/reader/decode/byte_rle.rs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    min_repeat_size: usize,
+    remaining: usize,
+}
+
+impl<R: Read> ByteRleIter<R> {
+    pub fn new(reader: R, length: usize) -> Self {
+        Self {
+            reader,
+            literals: [0u8; MAX_LITERAL_SIZE],
+            next_byte: None,
```
Comment (Jefffrey) [inline src/reader/decode/byte_rle.rs:25]: Is next_byte ever set anywhere?

File: src/reader/decode/byte_rle.rs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            num_literals: 0,
+            used: 0,
+            repeat: false,
+            min_repeat_size: MIN_REPEAT_SIZE,
```
Comment (Jefffrey) [inline src/reader/decode/byte_rle.rs:29]: nit: don't need struct field for const, as spec doesn't allow for this min_repeat_size to vary

**Comment by WenyXu** [comment]
@Jefffrey I mapped it to i8, PTAL

File: src/reader/decode/byte_rle.rs (hunk 1 above)
Comment (WenyXu) [reply src/reader/decode/byte_rle.rs]: Good point, I removed it.

**Review by Jefffrey — APPROVED** [review]
Thanks, looks good

**Pull request merged**